sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java

```diff
@@ -153,59 +171,73 @@ public HttpClient build() {
         OkHttpClient.Builder httpClientBuilder = this.okHttpClient == null
                 ? new OkHttpClient.Builder()
                 : this.okHttpClient.newBuilder();
-        //
+
+        // Add each interceptor that has been added.
         for (Interceptor interceptor : this.networkInterceptors) {
             httpClientBuilder = httpClientBuilder.addNetworkInterceptor(interceptor);
         }
-        if (this.readTimeout != null) {
-            httpClientBuilder = httpClientBuilder.readTimeout(this.readTimeout);
-        } else {
-            httpClientBuilder = httpClientBuilder.readTimeout(DEFAULT_READ_TIMEOUT);
-        }
-        if (this.connectionTimeout != null) {
-            httpClientBuilder = httpClientBuilder.connectTimeout(this.connectionTimeout);
-        } else {
-            httpClientBuilder = httpClientBuilder.connectTimeout(DEFAULT_CONNECT_TIMEOUT);
-        }
+
+        // Use the configured read timeout if set, otherwise use the default (120s).
+        httpClientBuilder = (this.readTimeout != null)
+            ? httpClientBuilder.readTimeout(this.readTimeout)
+            : httpClientBuilder.readTimeout(DEFAULT_READ_TIMEOUT);
```

Comment:
You might want to consider moving the ternary operation inside the method call and doing it on one line?

Reply:
Done